components/sidebar/sidebar_channel/sidebar_channel.tsx

```diff
@@ -121,6 +112,37 @@ export default class SidebarChannel extends React.PureComponent<Props, State> {
         };
     }
 
+    handleTransitionEnd = (event: TransitionEvent<HTMLLIElement>, draggableOnTransitionEnd?: TransitionEventHandler<HTMLLIElement>) => {
+        if (draggableOnTransitionEnd) {
+            draggableOnTransitionEnd(event);
+        }
+
+        // On fully collapsed
+        if (event.propertyName === 'height' && this.isCollapsed(this.props)) {
+            this.setState({show: false});
+        }
+
+        // On fully expanded
+        if (event.propertyName === 'height' && !this.isCollapsed(this.props)) {
+            this.setState({show: true});
```

Comment:
Yes i fixed it. I changed my approach to using the css 3 animation. This was done because
- browser can optimize for [animation steps](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Animations/Using_CSS_animations)
- animation start and end event listeners are typed correctly in react types, where as transition start doesn't exist in react dom types.